src/o365/spfx/commands/project/project-upgrade.ts

```diff
@@ -104,6 +119,31 @@ class SpfxProjectUpgradeCommand extends Command {
       return;
     }
 
+    let supportedTscVersion: { tscVersion: string, rushPackageVersion: string } = this.supportedTypeScriptVersions[0];
+
+    if (!!this.toTypeScriptVersion) {
+      const toVersionLessThan180 = this.toVersion < '1.8.0';
+      if (toVersionLessThan180) {
+        cb(new CommandError(`Versions of SPFx prior to 1.8.0 do not support upgrading TypeScript`, SpfxProjectUpgradeCommand.ERROR_UNSUPPORTED_TYPSCRIPT_VERSION));
+        return;
+      }
+
+      let isSupportedVersion: boolean = false;
```

Comment:
The following block could be simplified by using `this.supportedTypeScriptVersions.find()` which returns the object matching the specified predicate or undefined if no such object found.